src/guidelines.rst

```diff
@@ -70,8 +70,4 @@ Fortunately, there are options for optimizing the development of a package.
 Renaming Packages
 =================
 
-Sometimes packages are misnamed. To correct the name of the package, just
-change the name of the package in the ``recipe.yaml`` and rerender with
-``conda smithy rerender``. Note that this will not change the name of the repo
-and that the package with the old name will still exist. Please contact a
-member of conda-forge/core to remove the old package.
+Sometimes packages are misnamed. To correct the name of the package, please submit a PR into staged-recipes with the correct name. During the review process please make certain to note that the package is a rename and contact a member of conda-forge/core to remove the old package.
```

Comment:
tl;dr case-by-case basis is probably the best. As I was writing this up I see that @ocefpaf had a similar case-by-case comment.

@CJ-Wright In the proposed scenario, presumably those that are expecting spamlib when they go get spam will be surprised. Their package **may** fail to build or fail at runtime because their dependencies are incorrect. There are a few scenarios to consider.

1. (old-)spam has a higher version number than (new-)spam
   * **people that are packaging against (old-)spam:** Everything still works as expected for this group
   * **people that are packaging against (new-)spam:** They would need to pin to less than the version of (old-)spam
      
2. (old-)spam has a lower version number than (new-)spam
   * **people that are packaging against (old-)spam:** Their builds start breaking because (new-)spam is being picked up instead of (old-)spam. Difficult to track down this breakage unless you know about the package rename and if you know about the package rename than you can just change this to libspam or pin to the (old-)spam version. This is harmful to people who are non-expert packagers.
   * **people that are packaging against (new-)spam:** Everything works as expected.

3. Remove (old-)spam entirely. This is similar to scenario 2 except there is now no workaround for pinning to the last (old-)spam version and you're forced to migrate to libspam.